### ISSUE TITLE
autoexec table's 'path' column does not include filename when 'source' is services

### ISSUE THREAD

**Issue by jedwardsol** [osquery/osquery#8343]
# Bug report

Similar to https://github.com/osquery/osquery/issues/7262 which is fixed,  and relates to "drivers",  not services.

### What operating system and version are you using?

```
+------------+-------+----------+
| version    | build | platform |
+------------+-------+----------+
| 10.0.19045 | 19045 | windows  |
+------------+-------+----------+

```

### What version of osquery are you using?

```
+---------+
| version |
+---------+
| 5.12.1  |
+---------+

```

### What steps did you take to reproduce the issue?

```
osquery> select path  from services where name = "spooler";
+---------------------------------+
| path                            |
+---------------------------------+
| C:\Windows\System32\spoolsv.exe |
+---------------------------------+


osquery> select * from autoexec where source = "services" ;
<snip>
|                                                                              | spectrum                                 | services |
|                                                                              | Spooler                                  | services |
|                                                                              | sppsvc                                   | services |
<snip>
```

### What did you expect to see?

```
|  c:\windows\system32\spoolsv.exe                                      | Spooler                                  | services |
```

### What did you see instead?

```
|                                                                              | Spooler                                  | services |
```

This occurs for every service that isn't a svchost+servicedll based service

**Comment by directionless** [comment]
Reading the source code, I think the intent is as you expect.

https://github.com/osquery/osquery/blob/e712befdb211839c746631cc5180b7f02ed2ffef/osquery/tables/system/windows/autoexec.cpp#L22-L44

**Comment by pradishmp** [comment]
@seph
There are two ways to implement or create a Windows Services: one as a standalone executable and the other as a dynamic link library (DLL), also called the ServiceDLL. However, DLLs cannot run on their own; they require a host process to run, which is where **svchost.exe** comes into the picture. In the service table, the module_path column points to the path of this DLL.

The **svchost.exe** can host multiple service DLLs with related functionality, promoting resource efficiency. An example could be a service managing multiple network protocols, where each protocol is implemented in a separate DLL within **svchost.exe**, so this **svchost.exe** is a host for multiple services with related functionality implemented in a separate DLL.

to conclude we can say that if the service type is of "SHARE_PROCESS", only in those cases this module_path will have a valid path otherwise for all other types this value will be empty, I think this is what is happening in this case.

**Comment by michael-myers** [comment]
Yes, like the last commenter said, the original commenter is seeing services with no path because they are hosted in a "service host container" process. The only way to generate a path on disk for a service running in a `svchost.exe` is to do a lookup in the registry, at `HKLM:\SYSTEM\CurrentControlSet\Services\` under `Parameters\ServiceDLL`

I think exposing this information in osquery would be helpful, for quick detection of service DLLs loaded from unusual paths.

**Comment by pradishmp** [comment]
could you please check my slack comment (in **core** dated june 14th) on this issue, as i think the changes that i am proposing could be a breaking one.

**Comment by directionless** [comment]
@pradishmp I appreciate the detailed comment on slack. If I read that correctly correct, the change to the full arguments doesn't look useful. Is this something we can conditionalize based on service type?

**Comment by pradishmp** [comment]
> @pradishmp I appreciate the detailed comment on slack. If I read that correctly correct, the change to the full arguments doesn't look useful. Is this something we can conditionalize based on service type?

i will check on this and get back
